pkg/pulumiyaml/analyser.go

```diff
@@ -49,14 +51,63 @@ func (tc *typeCache) typeResource(r *runner, node resourceNode) bool {
 				Detail:      detail,
 				Subject:     subject,
 				Context:     &context,
-				Expression:  nil,
 				EvalContext: &hcl.EvalContext{},
 			})
 		} else {
 			tc.exprs[kvp.Value] = typ
 		}
 	}
 	tc.resources[node.Value] = hint.Fields()
+
+	options := ResourceOptionsTypeHint()
+	allOptions := make([]string, 0, len(options))
+	for k := range options {
+		allOptions = append(allOptions, k)
+	}
+	if s := v.Syntax(); s != nil {
+		if o, ok := s.(*syntax.ObjectNode); ok {
+			valid := []string{"type", "properties", "options", "condition", "metadata"}
+			fmtr := yamldiags.InvalidFieldBagFormatter{
+				ParentLabel: fmt.Sprintf("Resource %s", typ.String()),
+				MaxListed:   5,
+				Bags: []yamldiags.TypeBag{
+					{Name: "properties", Properties: allProperties},
+					{Name: "options", Properties: allOptions},
+					{Name: k, Properties: valid},
+				},
+				DistanceLimit: 3,
+			}
+			for i := 0; i < o.Len(); i++ {
+				prop := o.Index(i)
+				key := prop.Key.Value()
+				keyLower := strings.ToLower(key)
+				isValid := false
```

Comment:
Can just be `valid` or if we change `valid` above to something more explicit, e.g. `validKeys`.